chia/types/weight_proof.py

```diff
@@ -100,3 +103,63 @@ class WeightProof(Streamable):
     sub_epochs: List[SubEpochData]
     sub_epoch_segments: List[SubEpochChallengeSegment]  # sampled sub epoch
     recent_chain_data: List[HeaderBlock]
+
+
+@streamable
+@dataclass(frozen=True)
+class SubSlotDataV2(Streamable):
+    proof_of_space: Optional[ProofOfSpace]  # if challenge block
+    # VDF to signage point
+    cc_signage_point: Optional[VDFProof]  # if block
```

Comment:
what is the difference between `if challenge block`, and `if block`. Aren't we only verifying the challenge blocks?

Reply:
we are still validating vdfs from non challenge blocks in the case where its no blueboxed

Reply:
So are you saying that one subslot might be composed of many `SubSlotDataV2` objects, one for each block, if they are not blueboxed? If so we should put a comment for this

Reply:
yes this works the same way as the old proofs in that respect